### DISCUSSION TITLE
Submit Your Data Wrangling Challenges

### DISCUSSION THREAD

**Discussion by alapo** [alapo/Literate-Programming#15] [Q&A]
As part of the upcoming Module 07 I will be showcasing some typical data wrangling techniques. I'd like to invite anyone to submit problems they have had while importing datasets when it comes to data cleaning. I will try to incorporate all of the suggestions into one big "problematic dataset". 

Please make sure to describe the issue in detail and attach a file.

Deadline is Sunday at Midnight 

See you all soon
Andrew

**Comment by TinaSarkar** [comment]
Hey Andrew, 

I want to submit my data set on here, but the files are huge since it's sequencing data, not sure how else I can submit them to you.

Thanks, 

Tina

**Reply by alapo** [reply]
Hey Tina,

How huge are we talking?


You can submit part of the file rather than the whole file. The headers will remain the same.

**Reply by TinaSarkar** [reply]
Each zipped file is ~40MB and there are 96 files. Each file contains hundreds-thousand "reads" which are bacterial genomic sequences. If this is too complicated for the workshop I can just sit this one out.

**Reply by alapo** [reply]
Upload one file meow and I'll take a look.

**Reply by alapo** [reply]
There might be a max on GitHub. If that's the case submit some links (from Google Drive, OneDrive etc)

**Reply by alapo** [reply]
Hey Tina, I didnt get the file but I did think of you when I fell on this package which may be worth a read

https://github.com/vanheeringen-lab/seq2science

**Reply by TinaSarkar** [reply]
@alapo Thank you Andrew! I will take a look. Unfortunately the only way to transfer the files are through a physical hardrive it's far too large for Dropbox or Google Drive it would take days to upload.

Best,

Tina

**Comment by JoelBurma** [comment]
Hey Andrew, 

Attached is a file that can be used for the data wrangling/cleaning.

The primary issues are:
Channel 2 (Finapres - blood pressure): has spikes occurring within the data, which artificially raises the systolic data.
Channel 4 (Transcranial Doppler ultrasound - posterior cerebral artery blood velocity): the diastolic waveforms have a lot of dropouts occurring, so the diastolic values read inappropriately as 0 cm/s. This should be somewhat similar to trace in Channel 3.
Channel 5 (Electrocardiography - heart rate): The ECG waveform is inconsistent, where the R-spikes have varying amplitudes. This impacts the R-wave detection algorithms to properly detect the R-spikes.

Joel

[0.05_Squat_Stand.zip](https://github.com/alapo/Literate-Programming/files/6293174/0.05_Squat_Stand.zip)

**Comment by chelseachristie** [comment]
Hi Andrew,

Apologies for the late submission! 

Here is a file that could be used and some data wrangling issues. I attached a data dictionary too. Here is the code I used to start re-coding some of the variables:

# Start with loading packages

```{r load_packages, echo=FALSE}
# Load/Install required packages ---------------------
if (!require("pacman")) install.packages("pacman") # if pacman is not installed, the install it.
pacman::p_load(effectsize, ggeffects, ggpubr, ggstatsplot, googlesheets4, janitor, interactions, jtools, knitr, parameters, rio, remotes, report, rstatix, tidyverse,  see, sjPlot, sjmisc) # load/install required packages
```

# bring in the data & rename variable names

```{r load_data_&_rename, echo=FALSE}

# Bring in the data 
df <- rio::import("data/NC_birth_data_edited.xls")  


# rename variables where I don't like the original variable name

df %>%
  select(c("mage", "racemom", "gained", "drink", "tounces", "tgrams")) %>%
  rename(
    "mom_age_years" = "mage",
    "mom_race_cat" = "racemom",
    "mom_weight_gained" = "gained",
    "any_drinking" = "drink",
    "birth_weight_oz" = "tounces",
    "birth_weight_g" = "tgrams") # I think this worked - but why did it print it out afterwards? 
```

# re-coding variables

```{r recoding, echo=FALSE}

######################## re-coding variables ########################


##### sex of the infant
# I want to create a new variable here called "male_baby", that is created from the old variable called sex
# and then change the coding scheme so male = 1 & female = 0 (previously male = 1 and female = 2)


df$male_baby <- df$sex 

# I don't know what to do next... here is how I would do it in Stata:

# gen male_baby = sex
# recode male_baby (.=0) if sex == 2 
# recode male_baby (.=1) if sex == 1
# label define male_baby 0"girl" 1"boy"
# label value male_baby male_baby
# tab1 sex male_baby, m      # checking the result (checking that the frequencies match for the old and new variable)

# other variables that I might want to create, but don't know how:

# drink_or_smoke == a variable identifying mothers who smoked or drank alcohol during their pregnancy
# Chinese_mom == a variable where mothers who are Chinese and married == 1, and everybody else == 0
# above_avg_gain == a variable that indicates if mothers who gained more weight during their pregnancy compared to other women of their age category
# mom_weight_gain_truncated == a variable where mom_weight_gained values are truncated at 50 lbs (any values above 50 are transformed into 50)


[NC_birth_data_edited.xls](https://github.com/alapo/Literate-Programming/files/6298885/NC_birth_data_edited.xls)
[NC Birth Data_edited Info.docx](https://github.com/alapo/Literate-Programming/files/6298886/NC.Birth.Data_edited.Info.docx)

**Comment by TinaSarkar** [comment]
Thank you Andrew! I will take a look. Unfortunately the only way to transfer the files are through a physical hardrive it's far too large for Dropbox or Google Drive it would take days to upload.

Best,

Tina


Bachelor of Health Sciences (BHSc)

Cumming School of Medicine- University of Calgary

E: ***@***.***

P: 587-229-9611

________________________________
